### DISCUSSION TITLE
StdPrs_BRepFont - invalid conversion from 'unsigned char*'

### DISCUSSION THREAD

**Discussion by Escain** [Open-Cascade-SAS/OCCT#48] [General]
Hi,

I tried to build OCCT 7.8 on Windows with mingw, but got this compilation issue:

```
...src/StdPrs/StdPrs_BRepFont.cxx:460:30: error: invalid conversion from 'unsigned char*' to 'const char*' [-fpermissive]
[cmake]   460 |     const char* aTags      = &anOutline->tags[aStartIndex];
[cmake]       |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[cmake]       |                              |
[cmake]       |                              unsigned char*
```

It's probably enough to just `reinterpret_cast<const char*>(...)`.

Seems not resolved on master.
I hope it helps, otherwise you can just close this discussion.

**Comment by dpasukhi** [comment]
Hello. Thank you for your report and suggestion. Probably i will use your idea about casting, but needs to analyze the reasons.
I had the same issue on MacOS with FreeType. Found it only a few days ago.
The issue in FreeType lib and its methods. I will check. But if it reproducible in MinGW, it much helpful.

My test result on MacOS: https://github.com/dpasukhi/OCCT/actions/runs/10546249950/job/29217488781#step:5:5342

As a hot fix, you can disable freeType in CMake:  -`D USE_FREETYPE=OFF` (only in case if you don't need generate text as a shape)

@Escain could you share which version of FreeType do you use and how you receive that lib? (package manager, manual build, ...)
In my freetype's version tags field is `char*`.

Found a reason of changes: https://git.chiebot.com:10000/Chiebot-Mirror/freetype/commit/2a7bb4596f566a34fd53932af0ef53b956459d25?style=unified&whitespace=ignore-eol
Thank you. I will proceed with fix for master.

**Reply by dpasukhi** [reply]
PR: https://github.com/Open-Cascade-SAS/OCCT/pull/49

**Reply by Escain** [reply]
I built freetype from the latest release tag: https://github.com/freetype/freetype/tree/VER-2-13-3

Something like:

````
mkdir -p ../freetype
cmake -G Ninja  -DCMAKE_BUILD_TYPE=Release
                -DBUILD_SHARED_LIBS=OFF
                -DFT_WITH_ZLIB=OFF
                -DFT_WITH_BZIP2=OFF
                -DFT_WITH_PNG=OFF
                -DFT_WITH_HARFBUZZ=OFF
                ../freetype
````

thank you for the fix.

**Comment by dpasukhi** [comment]
Fixed by `0033808: Coding - FreeType Use unsigned point and contour indexing in FT_Outline` #49
